src/ip/QosConfig.cc

```diff
@@ -533,7 +533,11 @@ Ip::Qos::setSockTos(const int fd, tos_t tos, int type)
     // unless we pass it a 32-bit variable storing 8-bits of data.
     // NP: it is documented as 'int' for all systems, even those like Linux which accept 8-bit char
     //     so we convert to a int before setting.
+#if _SQUID_MINGW_
+    const char bTos = tos;
```

Comment:
Can you point to a document that defines the expected `IP_TOS` _value_ for MinGW to be a `char`? I am worried that we are fixing a compiler warning but adding a (silent) runtime bug here.

P.S. I understanding that using `const char` type for the underlying value changes the setsockopt() option value argument type from `int *` to `const char *` and fixes the compiler warning. However, the type of the underlying value is not determined by what _pointer_ type setsockopt() declaration expects. The same setsockopt() function works with all socket options, so its pointer type cannot determine the underlying option value type. We already had to fix the underlying type to match OS expectations in commit b4878a4. I want to make sure we are not re-introducing a problem that commit fixed.

Some Windows [documentation](https://learn.microsoft.com/en-us/windows/win32/winsock/ipproto-ip-socket-options) says that IP_TOS socket option is a DWORD boolean value (that should not be used), but I am _not_ suggesting that documentation is relevant to MinGW builds. I am only mentioning it here to illustrate the variety of underlying types that OSes might expect.

Reply:
Hmm, you are right. The DWORD is documented as a "boolean" type, which in turn is specified as "integer" sized. So we should just cast the pointer while leaving the type actually be `int`.

Updated in [9470fea](https://github.com/squid-cache/squid/pull/1359/commits/9470fea0f8df48a345b178ebe143e09805e701b3).

Reply:
Regarding the "Do not use" comment in their documentation, it then states that QoS API should be used to set DiffServ instead. However the QoS API referenced has been obsoleted and removed in Windows Vista. It looks like the `Socket Options` documentation is a bit outdated.